### PR TITLE
Modules via Github Packages - PR-2

### LINKED ISSUE #109: test: fix network-setup failed to normalize chaincode path
## To reproduce

Follow the steps here:
https://labs.hyperledger.org/weaver-dlt-interoperability/docs/external/getting-started/#fabric-network
Once you reach the part that says `make start-interop` it fails with this error:

```sh
Error: failed to normalize chaincode path: 'go list' failed with: go: github.com/golang/protobuf@v1.5.2: missing go.sum entry; to add it:
        go mod download github.com/golang/protobuf: exit status 1
!!!!!!!!!!!!!!! Chaincode packaging on peer0.org1 has failed !!!!!!!!!!!!!!!!

ERROR !!! Deploying chaincode failed
make: *** [Makefile:20: start-network1] Error 1
```

### Complete Log Output

[weaver-hl-lab-test-network-setup-chaincode-fail-2021-07-12.log](https://github.com/hyperledger-labs/weaver-dlt-interoperability/files/6804668/weaver-hl-lab-test-network-setup-chaincode-fail-2021-07-12.log)

### Operating System

```sh
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.2 LTS
Release:        20.04
Codename:       focal
```

### ProtoC Version

```sh
$ protoc --version
libprotoc 3.15.6
```

### Go Version

```sh
$ go version
go version go1.16.5 linux/amd64
```

### LINKED ISSUE #110: protos-go: grpc not compiled

### LINKED ISSUE #111: Modules via Github Packages
1. Compiled protos for JS, GO, Java, Kotlin.
2. Interoperation NodeJS SDK.
3. Corda Interop App.
4. Relay docker image.
5. Fabric Driver docker image.

### PR DESCRIPTION
**Pull request by sanvenDev**
Changes: 
1. Interoperation Cordapp package name changed to `com.weaver.corda.app.interop` and added steps to publish to github packages.
2. Modified all apps to use modules in github packages when not using `-local` in make commands. (Retained the functionality of make commands with `-local` suffix).
3. Modified go chaincodes to use previously published `protos-go`.
4. Interop-cc now being fetched as go module from github to install on testnet.

Close #111 .

Fixes:
1. Resolved #110 : Now compiled grpc go files are also created.
2. Resolved #109 : Added go.sum.

Comments before testing:
1. To start testnet there are two options:

- Fetch local copy: Uncomment line 45 and comment line 46 in `tests/network-setups/fabric/dev/Makefile`.
- Fetch remotely from my fork: Replace `hyperledger-labs` with `sanvenDev` in `line 7` and with `sanven\!dev` in `line 17`
in file: `tests/network-setups/fabric/dev/scripts/setupCC.sh`.

2. Replace `hyperledger-labs` with `sanvenDev` in `url` field in all `artifactory.properties`. (Currently I've published apps to my fork for testing, once approved I'll publish those in hyperledger-labs)

(Once this PR is merged, these steps won't be required)

### CODE REVIEW

File: common/protos-java-kt/README.md [hunk 1]
```diff
@@ -5,35 +5,27 @@
 1) Create a Personal Access Token from Github with write/read/delete access to packages.
```
Comment (VRamakrishna) [inline common/protos-java-kt/README.md:5]: Can you add more instructions here of the kind: Sign in to github.com - Select Settings --> Developer Settings --> Personal access tokens --> Generate new token

File: core/drivers/corda-driver/README.md [hunk 2]
```diff
@@ -51,6 +51,38 @@ The driver gRPC server will be listening on port `9099`.
...
+## Docker
+
+To build image, run:
+```
+make image
+```
+
+To push image to github container registry:
+
+* Create a Personal Access Token with write, read, and delete packages access in github.
```
Comment (VRamakrishna) [inline core/drivers/corda-driver/README.md:63]: More instructions would be good here, as I mentioned above.

File: core/drivers/corda-driver/gradle.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 kotlin.code.style=official
 name=corda-driver
-group=com.CordaDriver
-version=0.1
+group=com.weaver.corda.driver
+version=1.2.0
```
Comment (VRamakrishna) [inline core/drivers/corda-driver/gradle.properties:4]: Update to 1.2.1

File: core/drivers/corda-driver/makefile [hunk 4]
```diff
@@ -1,5 +1,15 @@
+DOCKER_IMAGE_NAME ?= weaver-corda-driver
+DOCKER_TAG ?= 1.2.0
```
Comment (VRamakrishna) [inline core/drivers/corda-driver/makefile:2]: Update to 1.2.1

File: core/network/corda-interop-app/README.md [hunk 5]
```diff
@@ -73,6 +48,39 @@ The docs are then located in `build/dokka/corda-interop-app`. Opening
 `index.html` in your browser will allow you to navigate through the project
 structure.
 
+## Steps to publish:
+
+1) Create a Personal Access Token from Github with write/read/delete access to packages.
```
Comment (VRamakrishna) [inline core/network/corda-interop-app/README.md:53]: More instructions here would be good, as I mentioned above.

File: core/relay/.env.template [hunk 6]
```diff
@@ -1,7 +1,7 @@
 PATH_TO_CONFIG=<PATH-TO-CONFIGURATION-FILE>
-RELAY_NAME=relay-network1
-RELAY_PORT=9080
-EXTERNAL_NETWORK=network1_net
+RELAY_NAME=<SAME_AS_IN_CONFIG>
```
Comment (VRamakrishna) [inline core/relay/.env.template:2]: Replace with <"name" in config toml file>

File: core/relay/.env.template [hunk 7]
```diff
@@ -1,7 +1,7 @@
...
+RELAY_PORT=<relay-server-port>
```
Comment (VRamakrishna) [inline core/relay/.env.template:3]: Replace with <relay-server-port/"port" in config toml file>

File: core/relay/relay-docker.md [hunk 8]
```diff
@@ -154,7 +154,7 @@ First **login** to github container registry:
 * Copy `.env.template` file to `.env`.
 * Keep following Environment Variables in `.env` and delete/ignore rest:
     * PATH_TO_CONFIG: Path to the relay's config file. e.g. `./config/Fabric_Relay.toml`.
-    * RELAY_NAME: Keep it same as in relay config file.
+    * RELAY_NAME: Keep it same as in relay config file. Same is used for container name.
```
Comment (VRamakrishna) [inline core/relay/relay-docker.md:157]: Say it's the "name" parameter in the config file.

File: core/relay/relay-docker.md [hunk 9]
```diff
@@ -169,14 +169,14 @@ First **login** to github container registry:
 
 * Copy `.env.template.2` file to `.env`.
 * Keep following Environment Variables in `.env` and delete/ignore rest:
-    * RELAY_NAME: Name for the relay.
+    * RELAY_NAME: Name for the relay. Same is used for container name.
```
Comment (VRamakrishna) [inline core/relay/relay-docker.md:172]: As above.

File: samples/fabric/fabric-cli/src/commands/interop-helper.ts [hunk 10]
```diff
@@ -151,8 +151,9 @@ const command: GluegunCommand = {
           interopFlowResponse.views[0].toObject()
         )}. Interop Flow result: ${interopFlowResponse.result || 'successful'}`
       )
+      const remoteValue = InteroperableHelper.getResponseDataFromView(interopFlowResponse.views[0])
       spinner.succeed(
-        `Called Function ${applicationFunction}. With Args: ${invokeObject.ccArgs} `
+        `Called Function ${applicationFunction}. With Args: ${invokeObject.ccArgs} ${remoteValue}`
```
Comment (VRamakrishna) [inline samples/fabric/fabric-cli/src/commands/interop-helper.ts:156]: Add comma before ${remoteValue}

File: tests/network-setups/corda/README.md [hunk 11]
```diff
@@ -49,19 +49,20 @@ required CorDapps are in
...
 respective repos.
 
-To get the CorDapps from Artifactory you will need to have permission to do so
-. Copy the `artifactory.properties.template`
-file as `artifactory.properties` and replace placeholders with your IBM email
-and Artifactory API key (found on your profile at
-https://na.artifactory.swg-devops.com/artifactory/).
+To get the CorDapps from Github Packages you will need to have permission to do so:
+
+1) Create a Personal Access Token from Github with read access to packages.
```
Comment (VRamakrishna) [inline tests/network-setups/corda/README.md]: More instructions would be good here as I mentioned above.

File: common/protos-java-kt/README.md [hunk 12]
```diff
@@ -5,35 +5,27 @@
...
 1) Create a Personal Access Token from Github with read access to packages.
 2) Create a copy of `artifactory.properties.template` as `artifactory.properties`.
 3) Replace <GITHUB Email> with your email id for github.
-3) Replace <GITHUB Personal Access Token> with your personal access token.
-4) Add this to your build.gradle (change the version accordingly):
+4) Replace <GITHUB Personal Access Token> with your personal access token.
+5) Add this to your build.gradle (change the version accordingly):
 ```
 dependencies {
 	compile(group: 'com.weaver', name: 'protos-java-kt', version: "1.2.0")
```
Comment (VRamakrishna) [inline common/protos-java-kt/README.md:31]: Update version to 1.2.0

File: core/network/corda-interop-app/README.md [hunk 13]
```diff
@@ -73,6 +48,39 @@ The docs are then located in `build/dokka/corda-interop-app`. Opening
...
+## Steps to use module published in Github Packages with Gradle:
+
+1) Create a Personal Access Token from Github with read access to packages.
+2) Create a copy of `artifactory.properties.template` as `artifactory.properties`.
+3) Replace <GITHUB Email> with your email id for github.
+4) Replace <GITHUB Personal Access Token> with your personal access token.
+5) Add this to your build.gradle (change the version accordingly):
+```
+dependencies {
+	compile(group: 'com.weaver.corda.app.interop', name: 'interop-contracts', version: "1.2.0")
```
Comment (VRamakrishna) [inline core/network/corda-interop-app/README.md:79]: Update version on this line and the one below to 1.2.1

File: core/relay/relay-docker.md (hunk 9 above)
Comment (sanvenDev) [reply core/relay/relay-docker.md:172]: Hi Rama, this set of instructions doesn't involve any config file. (It gets generated). So not sure if same above comment would apply here?
Here we are explicitly setting the Relay Name for the first time.

File: samples/fabric/fabric-cli/src/commands/interop-helper.ts (hunk 10 above)
Comment (sanvenDev) [reply samples/fabric/fabric-cli/src/commands/interop-helper.ts:156]: Comma is already there when printing ${invokeObject.ccArgs}, as thats a list.

File: common/protos-java-kt/README.md (hunk 12 above)
Comment (sanvenDev) [reply common/protos-java-kt/README.md:31]: As this is an example, I had a doubt about this should we upgrade the Readme example everytime there's a release?
In Instructions I've specified to change the version as per requirements.

**Review by VRamakrishna — APPROVED** [review]

**Pull request merged**